symfony/client/src/orangehrmPimPlugin/pages/customField/EditCustomField.vue

```diff
@@ -0,0 +1,210 @@
+          <oxd-button
+            type="button"
+            displayType="ghost"
+            label="Cancel"
+            @click="onCancel"
+          />
+          <submit-button />
+        </oxd-form-actions>
+      </oxd-form>
+    </div>
+  </div>
+</template>
+
+<script>
+import {navigate} from '@orangehrm/core/util/helper/navigation';
+import {APIService} from '@orangehrm/core/util/services/api.service';
+import {
+  required,
+  shouldNotExceedCharLength,
+} from '@/core/util/validation/rules';
+
+const customFieldModel = {
+  fieldName: '',
+  screen: '',
+  fieldType: '',
+  extraData: '',
+};
+
+export default {
+  props: {
+    customFieldId: {
+      type: Number,
+      required: true,
+    },
+    screenList: {
+      type: Array,
+      required: true,
+    },
+    fieldTypeList: {
+      type: Array,
+      required: true,
+    },
+  },
+
+  data() {
+    return {
+      isLoading: false,
+      customField: {...customFieldModel},
+      rules: {
+        fieldName: [required, shouldNotExceedCharLength(250)],
+        screen: [shouldNotExceedCharLength(100)],
+        fieldType: [required, shouldNotExceedCharLength(15)],
+        extraData: [shouldNotExceedCharLength(250)],
+      },
+    };
+  },
+
+  setup() {
+    const http = new APIService(
+      window.appGlobal.baseUrl,
+      '/api/v2/pim/custom-fields',
+    );
+    return {
+      http,
+    };
+  },
+
+  methods: {
+    onSave() {
+      this.isLoading = true;
+      const fieldType = this.customField.fieldType.map(item => item.id)[0];
+      this.http
+        .update(this.customFieldId, {
+          fieldName: this.customField.fieldName,
+          screen: this.customField.screen.map(item => item.id)[0],
+          fieldType: fieldType,
+          extraData: fieldType === 1 ? this.customField.extraData : null,
+        })
+        .then(() => {
+          return this.$toast.updateSuccess();
+        })
+        .then(() => {
+          this.onCancel();
+        });
+    },
+    onCancel() {
+      navigate('/pim/listCustomFields');
+    },
+  },
+  created() {
+    this.isLoading = true;
+    this.http
+      .get(this.customFieldId)
+      .then(response => {
+        const {data} = response.data;
+        this.customField.fieldName = data.fieldName;
+        if (data.screen !== '' && data.screen !== null) {
+          this.customField.screen = [
+            this.screenList.find(c => {
+              return c.id === data.screen;
+            }),
+          ];
+        }
+        if (data.fieldType !== '' && data.fieldType !== null) {
+          this.customField.fieldType = [
+            this.fieldTypeList.find(c => {
+              return c.id === data.fieldType;
+            }),
+          ];
+        }
+        this.customField.extraData = data.extraData;
+      })
+      .finally(() => {
+        this.isLoading = false;
+      });
+  },
+
+  computed: {
+    isDropDownField() {
+      // eslint-disable-next-line vue/no-side-effects-in-computed-properties
```

Comment:
are sure about this. @Super-Chama can you check

Reply:
removed